src/poll.rs

```diff
@@ -56,14 +90,72 @@ fn emit_event(event_handler: &Mutex<dyn EventHandler>, res: Result<Event>) {
     }
 }
 
+impl PathData {
+    pub fn collect<BH: BuildHasher>(
+        path: &Path,
+        metadata: &Metadata,
+        build_hasher: Option<&BH>,
+        last_check: Instant,
+    ) -> Self {
+        let mtime = FileTime::from_last_modification_time(metadata).seconds();
+        let hash = metadata
+            .is_file()
+            .then(|| build_hasher.and_then(|bh| Self::hash_file(path, bh).ok()))
+            .flatten();
+        Self {
+            mtime,
+            hash,
+            last_check,
+        }
+    }
+
+    fn hash_file<P: AsRef<Path>, BH: BuildHasher>(path: P, build_hasher: &BH) -> io::Result<u64> {
+        let mut hasher = build_hasher.build_hasher();
+        let mut file = fs::File::open(path)?;
+        let mut buf = [0; 512];
+        while file.read(&mut buf)? > 0 {
+            hasher.write(&buf);
+        }
+        Ok(hasher.finish())
+    }
+
+    pub fn detect_change(&self, other: &PathData) -> Option<EventKind> {
+        if self.mtime > other.mtime {
+            Some(EventKind::Modify(ModifyKind::Metadata(
+                MetadataKind::WriteTime,
+            )))
+        } else if self.hash != other.hash {
+            Some(EventKind::Modify(ModifyKind::Data(DataChange::Any)))
+        } else {
+            None
+        }
+    }
+}
+
 impl PollWatcher {
     /// Create a new [PollWatcher] and set the poll frequency to `delay`.
+    #[deprecated(note = "Prefer with_config instead")]
```

Comment:
I'd rather drop this directly, this is a preview - so we can change stuff as we like.

Reply:
(reason: if we stabilize this, we could only remove it in v6, whenever that is)